src/views/webview-app/components/connect-form/general-tab/host/host.tsx

```diff
@@ -247,7 +261,7 @@ export class HostInput extends React.PureComponent<StateProps & DispatchProps> {
               placeholder="27017"
               changeHandler={(event: React.ChangeEvent<HTMLInputElement>): void => this.onHostChanged({
                 host: host.host,
-                port: event.target.value
+                port: +event.target.value.trim()
```

Comment:
isn't `parseInt` more clear?

Reply:
`parseInt()` would also accept values like `123 abc` … we don’t want that here, do we?

Reply:
Actually `+event.target.value.trim()` is repeated quite a number of time, maybe a function like `parsePortFromEvent` would be nice

Reply:
I think the trim here isn't actually needed since html's `number` input handles that for us. I'll remove that. I don't think we need a helper than for just having the `+`. Would be nice if typescript could pull types from the input type.

Reply:
> parseInt() would also accept values like '123 abc' … we don’t want that here, do we?

Right + is even better, do we need to account for `NaN` though, i believe we have some validation though

Reply:
I don't think we need to. I don't think there's any easy way to get to `NaN` in an html number input field. I think it's probably best we don't perform much validation on the form level unless its something we really know the user probably doesn't want to do.

Reply:
Makes sense!

Reply:
> we don't perform much validation on the form level

I mean having a type=number field is the best kind of validation we can do there so is not possible to input any sort of bad value